tools/autogen

```diff
@@ -16,7 +16,7 @@
 # Remove the autom4te.cache directory to make sure we start in a clean state
 rm -rf autom4te.cache
 
-autoconf --force --warnings=all,error
+${1-autoconf} --force --warnings=all,error
```

Comment:
Apropos the comment on clarity below, is this any better?

```suggestion
# Default to autoconf if no parameter passed
AUTOCONF_TOOL="${1-autoconf}"
$AUTOCONF_TOOL --force --warnings=all,error
```

Reply:
Yes, it's better. This being said, I did guess on my own what `${1-autoconf}` could possibly mean when reading the patch, before seeing the callsite. So either it's sort of clear enough to be guessable by novices, or my brain is getting shell-programming-damaged without me noticing.